tfhe/src/core_crypto/algorithms/lwe_linear_algebra.rs

```diff
@@ -235,20 +279,51 @@ pub fn lwe_ciphertext_plaintext_add_assign<Scalar, InCont>(
 ) where
     Scalar: UnsignedInteger,
     InCont: ContainerMut<Element = Scalar>,
+{
+    let ciphertext_modulus = lhs.ciphertext_modulus();
+    if ciphertext_modulus.is_compatible_with_native_modulus() {
+        lwe_ciphertext_plaintext_add_assign_native_mod_compatible(lhs, rhs);
+    } else {
+        lwe_ciphertext_plaintext_add_assign_other_mod(lhs, rhs);
+    }
+}
+
+pub fn lwe_ciphertext_plaintext_add_assign_native_mod_compatible<Scalar, InCont>(
+    lhs: &mut LweCiphertext<InCont>,
+    rhs: Plaintext<Scalar>,
+) where
+    Scalar: UnsignedInteger,
+    InCont: ContainerMut<Element = Scalar>,
 {
     let body = lhs.get_mut_body();
     let ciphertext_modulus = body.ciphertext_modulus();
     assert!(ciphertext_modulus.is_compatible_with_native_modulus());
-    if ciphertext_modulus.is_native_modulus() {
-        *body.data = (*body.data).wrapping_add(rhs.0);
-    } else {
-        *body.data = (*body.data).wrapping_add(
-            rhs.0
-                .wrapping_mul(ciphertext_modulus.get_power_of_two_scaling_to_native_torus()),
-        );
+    match ciphertext_modulus.kind() {
+        CiphertextModulusKind::Native => *body.data = (*body.data).wrapping_add(rhs.0),
```

Comment:
Could factor `*body.data = (*body.data).wrapping_add(` out of the `match`

Reply:
IMO
```
let scaled_rhs =  if ciphertext_modulus.is_native_modulus() {
    rhs.0
} else {
    rhs.0.wrapping_mul(ciphertext_modulus.get_power_of_two_scaling_to_native_torus())
};

*body.data = (*body.data).wrapping_add(scaled_rhs);
```
is cleaner

Reply:
Sure, a match seems better
But I still think it's better to have the assignment out of the conditional